src/Domain/Creators/Python/BinaryCompactObject.cpp

```diff
@@ -0,0 +1,69 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Creators/Python/BinaryCompactObject.hpp"
+
+#include <array>
+#include <cstddef>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <vector>
+
+#include "Domain/CoordinateMaps/Distribution.hpp"
+#include "Domain/Creators/BinaryCompactObject.hpp"
+#include "Domain/Creators/DomainCreator.hpp"
+
+namespace py = pybind11;
+
+namespace domain::creators::py_bindings {
+void bind_binary_compact_object(py::module& m) {
+  py::class_<domain::creators::BinaryCompactObject<false>, DomainCreator<3>>(
+      m, "BinaryCompactObject")
+      .def(py::init(
+               [](const double inner_radius_object_a,
+                  const double outer_radius_object_a, const double x_coord_a,
+                  const bool excise_a, const bool use_logarithmic_map_a,
+                  const double inner_radius_object_b,
+                  const double outer_radius_object_b, const double x_coord_b,
+                  const bool excise_b, const bool use_logarithmic_map_b,
+                  const std::array<double, 2> center_of_mass_offset,
+                  const double envelope_radius, const double outer_radius,
+                  const double cube_scale, const size_t initial_refinement,
+                  const size_t initial_num_points,
+                  const bool use_equiangular_map,
+                  const std::vector<double>& radial_partitioning_outer_shell,
+                  const double opening_angle_in_degrees) {
+                 return domain::creators::BinaryCompactObject<false>{
+                     domain::creators::BinaryCompactObject<false>::Object{
+                         inner_radius_object_a, outer_radius_object_a,
+                         x_coord_a, excise_a, use_logarithmic_map_a},
+                     domain::creators::BinaryCompactObject<false>::Object{
+                         inner_radius_object_b, outer_radius_object_b,
+                         x_coord_b, excise_b, use_logarithmic_map_b},
+                     center_of_mass_offset,
+                     envelope_radius,
+                     outer_radius,
+                     cube_scale,
+                     initial_refinement,
+                     initial_num_points,
+                     use_equiangular_map,
+                     domain::CoordinateMaps::Distribution::Logarithmic,
+                     radial_partitioning_outer_shell,
+                     domain::CoordinateMaps::Distribution::Linear,
+                     opening_angle_in_degrees,
+                     std::nullopt};
```

Comment:
Why provide one default option explicitly here but not others?

Reply:
That is a good question, I'll take it out